subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -65,50 +65,40 @@ static inline CborError
 os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
 				    const struct k_thread *thread)
 {
-	int err = 0;
-
-	ARG_UNUSED(idx);
-
-	err = cbor_encode_text_string(encoder, thread->name, strnlen(thread->name,
-				      CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1));
-	return err;
-}
-
-#elif defined(CONFIG_OS_MGMT_TASKSTAT_USE_THREAD_PRIO_FOR_NAME)
-static inline CborError
-os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
-				    const struct k_thread *thread)
-{
-	CborError err = 0;
-	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+	size_t name_len = strlen(thread->name);
 
 	ARG_UNUSED(idx);
 
-	thread_name[sizeof(thread_name) - 1] = 0;
-	ll_to_s((int)thread->base.prio, sizeof(thread_name) - 1, thread_name);
+	if (name_len > (CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN)) {
```

Comment:
minor nit.

`CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN` is defined as an int, so you may remove the extra set of parenthesis.
https://github.com/zephyrproject-rtos/zephyr/blob/f2dde2819f8fed317ec7cc9432708db7609b0da6/subsys/mgmt/mcumgr/Kconfig#L274-L275

```suggestion
	if (name_len > CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN) {
```